orca-core/src/main/java/com/netflix/spinnaker/orca/pipeline/DefaultStageDefinitionBuilderFactory.java

```diff
@@ -40,6 +43,9 @@ public DefaultStageDefinitionBuilderFactory(StageDefinitionBuilder... stageDefin
       .stream()
       .filter((it) -> it.getType().equals(stage.getType()) || it.getType().equals(stage.getContext().get("alias")))
       .findFirst()
-      .orElseThrow(() -> new NoSuchStageDefinitionBuilder(stage.getType()));
+      .orElseThrow(() -> {
+        List<String> knownTypes = stageDefinitionBuilders.stream().map(it -> it.getType()).collect(toList());
```

Comment:
Maybe throw a `.sorted()` in before the `.collect` to improve debugging?

When I run this locally, I have 135 stage definition builders, and they are in a kind of random order.